### ISSUE TITLE
version number incompatible with PEP 440

### ISSUE THREAD

**Issue by sveneh** [ethereum/pyethereum#193]
When I do a fresh build with buildout, I get warning about the version. The warning is not critical, just cosmetic.

the warning is generated by setuptools-10.2. Build is done here in docker.

```
 RUN cd pyethereum && curl https://bootstrap.pypa.io/bootstrap-buildout.py | python
 ---> Running in b35db5de9573
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  6501  100  6501    0     0  18631      0 --:--:-- --:--:-- --:--:-- 18681
Downloading https://pypi.python.org/packages/source/s/setuptools/setuptools-10.2.zip
Extracting in /tmp/tmpQnH4ue
Now working in /tmp/tmpQnH4ue/setuptools-10.2
Building a Setuptools egg in /tmp/tmpJynlZV
/tmp/tmpJynlZV/setuptools-10.2-py2.7.egg
Creating directory '/pyethereum/bin'.
Creating directory '/pyethereum/parts'.
Creating directory '/pyethereum/eggs'.
Creating directory '/pyethereum/develop-eggs'.
Generated script '/pyethereum/bin/buildout'.
 ---> 6f14d09512a9
Removing intermediate container b35db5de9573
Step 3 : RUN cd pyethereum && bin/buildout
 ---> Running in ca7d93e6deed
Develop: '/pyethereum/.'
/pyethereum/eggs/setuptools-10.2-py2.7.egg/setuptools/dist.py:293: UserWarning: The version specified ('23c7d1e') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
```

TL;DR: version ('23c7d1e') is not a valid python version, according to https://www.python.org/dev/peps/pep-0440/

**Comment by sveneh** [comment]
I see the version string has changed, but it is still not compatible.
```
/pyethereum/eggs/setuptools-12.0.5-py2.7.egg/pkg_resources/__init__.py:2510: PEP440Warning: 'pyethereum (0.7.51-40-g8d7d3be)' is being parsed as a legacy, non PEP 440, version. You may find odd behavior and sort order. In particular it will be sorted as less than 0.0. It is recommend to migrate to PEP 440 compatible versions.
```

**Comment by heikoheiko** [comment]
depends on: https://github.com/warner/python-versioneer/pull/45
